Proteomics/ProteolyticDigestion/PeptideWithSetModifications.cs

```diff
@@ -201,13 +220,19 @@ public IEnumerable<Product> Fragment(DissociationType dissociationType, Fragment
 
                             for (int n = f; n < terminalMasses.Length; n++)
                             {
-                                yield return new Product(productType, terminalMasses[n], neutralLoss);
+                                if (skippers.Count == 0)
```

Comment:
this skippers stuff is problematic for a couple of reasons with regards to z+1 ions and proline:
- The location of P is determined by its position in the FullSequence; I think it should be in the base sequence
- The index in the array of terminal masses is not the position in the base sequence

When I look at the unit test for the no proline Z fragments (of PEPTIDE), I'm missing ion 5 when I should be missing ions 1 and 3